addons/persistence/infinispan-persistence-addon/src/main/java/org/kie/kogito/infinispan/ProtoStreamObjectMarshallingStrategy.java

```
@@ -91,6 +96,28 @@ public void registerMarshaller(MessageMarshaller<?>... marshallers) {
             typeToClassMapping.putIfAbsent(marshaller.getTypeName(), marshaller.getJavaClass());
         }
     }
+    
+    @Override
+	 public String marshalToJson(Object object) {
+		String json = null;
+		try {
+			json = MAPPER.writeValueAsString(object);
+		} catch (JsonProcessingException e) {
+			 LOGGER.error("Error while writing object as json", e);
+		}
+		return json;
+  	 
+   }
+
+	@Override
+	public Object unmarshlFromJson(String dataType, String json) {
+		try {
+			return MAPPER.readValue(json, serializationContext.getMarshaller(dataType).getJavaClass());
+		} catch (JsonProcessingException e) {
+			 LOGGER.error("Error while reading object from json", e);
```

I think it should throw an exception instead of just logging the error.